README.md

```diff
@@ -17,24 +17,40 @@ For more information, visit the CrownLabs website ([https://crownlabs.polito.it]
 
 ## Architecture
 
-CrownLabs relies on two major components:
+CrownLabs relies on two major high-level components:
+* **The Backend Business Logic**, which provides the different CrownLabs functionalities and is implemented by custom Kubernetes operators (e.g. the Instance Operator);
+* **The Frontend Dashboard**, which interacts with the Kubernetes API Server and exposes the different CrownLabs custom resources through a graphical interface.
 
-* **Frontend**, which is responsible to access Kubernetes API, guiding the user to creation of VMs.
-* **Laboratory Operator**, which reacts to LabInstances creation by creating the Kubernetes objects to launch
-the laboratory.
-
-A high-level representation of the main architectural building blocks composing CrownLabs is given by the following figure. Please notice that, for the sake of clarity, the figure depicts the elements essential for the provision of the actual service (i.e. remote computing labs), while leaving out with those more low-level or associated with the cluster operation (e.g monitoring).
+A high-level representation of the main architectural building blocks composing CrownLabs is given by the following figure.
+Please notice that, for the sake of clarity, the figure depicts the elements essential for the provision of the actual service (i.e. remote computing labs), while leaving out with those more low-level or associated with the cluster operation (e.g monitoring).
 
 ![CrownLabs High-Level Architecture](documentation/architecture.svg)
 
+## Backend Business Logic
+
+The backend business logic providing the different CrownLabs functionalities is implemented by custom Kubernetes operators, while the data model is defined by means of CRDs.
+Specifically, the main backend components are:
+
+* the Instance Operator, which implements the logic to spawn new environments starting from predefined templates;
+* the Tenant Operator, which automates the management of CrownLabs users (i.e. tenants) and groups (i.e. workpaces);
+* the Bastion Operator, which configures an SSH bastion to provide command-line access to the environments.
```

Comment:
```suggestion
* the **Bastion Operator**, which configures an SSH bastion to provide command-line access to the environments instead of the web-based GUI.
```